app/services/foreman_ansible/inventory_creator.rb

```diff
@@ -8,11 +8,42 @@ def initialize(hosts)
       @hosts = hosts
     end
 
-    def tempfile
-      tempfile = Tempfile.new("foreman-#{SecureRandom.uuid}-inventory")
-      tempfile.write(hosts.map(&:fqdn).join('\n'))
-      tempfile.close
-      tempfile
+    def json
+      { 'all' => { 'hosts' => hosts.map(&:fqdn) },
+        '_meta' => { 'hostvars' => host_vars } }
+    end
+
+    def host_vars
+      hosts.reduce({}) do |hash, host|
+        hash.update(host.fqdn => { 'foreman' => host_attributes(host),
+                                   'foreman_params' => host_params(host),
+                                   'foreman_ansible_roles' => host_roles(host),
+                                   'foreman_facts' => host_facts(host) })
+      end
+    end
+
+    def host_roles(host)
+      host.all_ansible_roles.map(&:name)
+    end
+
+    def host_attributes(host)
+      render_rabl(host, 'api/v2/hosts/main')
+    end
+
+    def host_params(host)
+      render_rabl(host.host_parameters.authorized, 'api/v2/parameters/base')
```

Comment:
Call `host.host_params` instead. The reasons are two:

* `host.host_parameters` does NOT contain hostgroup parameters (or any inherited parameters)
* `host.host_params` provides params in JSON already, so you don't have to render anything